src/Tizen.Diagnostics/Tizen.Diagnostics/Diagnostics.cs

```diff
@@ -0,0 +1,298 @@
+/*
+* Copyright (c) 2020 Samsung Electronics Co., Ltd All Rights Reserved
+*
+* Licensed under the Apache License, Version 2.0 (the License);
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using NativeApi = Interop.Diagnostics;
+
+namespace Tizen.Diagnostics
+{
+    /// <summary>
+    /// The Diagnostics class is used for requesting client data and receiving diagnostics report events.
+    /// </summary>
+    /// <since_tizen> 8 </since_tizen>
+    public static class Diagnostics
+    {
+        private static NativeApi.NotificationCallback _notificationCallback;
+        private static event EventHandler<ReportReceivedEventArgs> _reportReceived;
+
+        /// <summary>
+        /// Event to be invoked when new report context arrives.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        public static event EventHandler<ReportReceivedEventArgs> ReportReceived
+        {
+            add
+            {
+                if (_reportReceived == null)
+                    SetNotificationCallback();
+                _reportReceived += value;
+            }
+            remove
+            {
+                _reportReceived -= value;
+                if (_reportReceived == null)
+                    UnsetNotificationCallback();
+            }
+        }
+
+        /// <summary>
+        /// Requests other apps to dump diagnostics data.
+        /// </summary>
+        /// <since_tizen> 8 </since_tizen>
+        /// <param name="clientId">An id of app or service to request.</param>
+        /// <param name="parameters">Array of parameters passed to the client.</param>
+        /// <returns>
+        /// The instance of DiagnosticsData.
+        /// </returns>
+        /// <privilege>
+        /// </privilege>
+        /// <remarks>
+        /// This function is permitted only to an app signed by platform level certificates.
+        /// </remarks>
+        public static DiagnosticsData RequestClientData(string clientId, string[] parameters)
```

Comment:
Can't users choose the report type of `DiagnosticsData`? Is the different kind of data with the data from Received event?

Reply:
`DiagnosticsData` is just a stream, it is not related to any type of report. It can be data from any report requested with `DiagnosticsContext.GetReport()` or data from report requested from other apps with `Diagnostics.RequestClientData()`

Reply:
Actually `ReportReceived` event does not provide any specific report data yet. It just provides report context, from where user can get specific type of the report. So user must call `GetReport` in the event handler to get the report. Generated event means something like this: "Hi, I'm crash worker, some app has just crashed and I have some reports to offer. What kind of report do you want?"